src/app/compressed-air-assessment/end-uses/end-uses.service.ts

```diff
@@ -189,7 +211,7 @@ export class EndUsesService {
       }
     }
 
-    return endUseEnergyData;
+    return {hasValidEndUses: !hasInvalidEndUse, endUseEnergyData};
```

Comment:
I don't think that this is correct. I think it should be:

`{
  hasValidEndUses: !hasInvalidEndUses, 
  endUseEnergyData: endUseEnergyData
}`

as is I think endUseEnergyData will always be undefined.